src/App.tsx

```diff
@@ -17,79 +22,125 @@ export const goodsFromServer = [
-  console.log(sortType, isReversed);
+      default:
+        return 0;
+    }
+  });
 
-  return visibleGoods;
+  if (isReversed) {
+    goodsToShow.reverse();
+  }
+
+  return goodsToShow;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+export class App extends Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
 
-      <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
-      </ul>
-    </div>
-  );
-};
+  sortByLength = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  sortByAlphabet = () => {
+    this.setState({ sortType: SortType.ALPHABET });
+  };
+
+  reverse = () => {
+    this.setState((state) => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  reset = () => {
+    this.setState({
+      sortType: SortType.NONE,
+      isReversed: false,
+    });
+  };
+
+  render() {
+    const { sortType, isReversed } = this.state;
+    const reorderedGoods = getReorderedGoods(goodsFromServer, this.state);
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <Button
+            onClick={this.sortByAlphabet}
+            className={
+              classNames('is-info', {
+                'is-light': sortType !== SortType.ALPHABET,
+              })
+            }
+          >
+            Sort alphabetically
+          </Button>
+
+          <Button
+            onClick={this.sortByLength}
+            className={
+              classNames('is-success', {
+                'is-light': sortType !== SortType.LENGTH,
+              })
+            }
+          >
+            Sort by length
+          </Button>
+
+          <Button
+            onClick={this.reverse}
+            className={
+              classNames('is-warning', {
+                'is-light': !isReversed,
+              })
+            }
+          >
+            Reverse
+          </Button>
+
+          {(sortType !== SortType.NONE || isReversed) && (
```

Comment:
better to create separate variable for that
If you will need to add one more condition later - you will have to find it in your code. Will be much easier if it is stored inside the separate variable